packages/teleterm/src/ui/TabHost/TabHost.tsx

```diff
@@ -28,22 +28,19 @@ import { ClusterConnectPanel } from './ClusterConnectPanel/ClusterConnectPanel';
 export function TabHostContainer() {
   const ctx = useAppContext();
   ctx.workspacesService.useState();
-
   const isRootClusterSelected = !!ctx.workspacesService.getRootClusterUri();
 
-  return useMemo(() => {
-    if (isRootClusterSelected) {
-      return <TabHost />;
-    }
-    return <ClusterConnectPanel />;
-  }, [isRootClusterSelected]);
```

Comment:
No, `useMemo` wasn't useful at all, because we were listening for the same state updates in both parent and child anyway. After the changes, we will subscribe only in the parent (which will re-render child) to make things simpler.